packages/nx/src/plugins/js/project-graph/build-dependencies/target-project-locator.spec.ts

```diff
@@ -1,776 +1,763 @@
-    expect(res4).toEqual('proj');
-  });
-
-  it('should be able to resolve local project', () => {
-    jest
-      .spyOn(targetProjectLocator as any, 'resolveImportWithRequire')
-      .mockReturnValue('libs/proj1/index.ts');
-
-    const result1 = targetProjectLocator.findProjectWithImport(
-      '@org/proj1',
-      'libs/proj1/index.ts'
-    );
-    expect(result1).toEqual('@org/proj1');
-
-    jest
-      .spyOn(targetProjectLocator as any, 'resolveImportWithRequire')
-      .mockReturnValue('libs/proj1/some/nested/file.ts');
-    const result2 = targetProjectLocator.findProjectWithImport(
-      '@org/proj1/some/nested/path',
-      'libs/proj1/index.ts'
-    );
-    expect(result2).toEqual('@org/proj1');
-  });
-
-  it('should be able to npm dependencies', () => {
-    const result1 = targetProjectLocator.findProjectWithImport(
-      '@ng/core',
-      'libs/proj1/index.ts'
-    );
-    const result2 = targetProjectLocator.findProjectWithImport(
-      'npm-package',
-      'libs/proj1/index.ts'
-    );
-
-    expect(result1).toEqual('npm:@ng/core');
-    expect(result2).toEqual('npm:npm-package');
-  });
+jest.mock('@nx/devkit', () => ({
+  ...jest.requireActual<any>('@nx/devkit'),
+  workspaceRoot: '/root',
+}));
 
-  it('should be able to resolve paths that have similar names', () => {
-    const proj = targetProjectLocator.findProjectWithImport(
-      '@proj/proj123',
-      ''
-    );
-    expect(proj).toEqual('proj123');
-
-    const childProj = targetProjectLocator.findProjectWithImport(
-      '@proj/proj1234-child',
-      ''
-    );
-    expect(childProj).toEqual('proj1234-child');
-
-    const parentProj = targetProjectLocator.findProjectWithImport(
-      '@proj/proj1234',
-      ''
-    );
-    expect(parentProj).toEqual('proj1234');
-  });
+jest.mock('nx/src/utils/workspace-root', () => ({
+  workspaceRoot: '/root',
+}));
 
-  it('should be able to resolve npm projects', () => {
-    const similarImportFromNpm = targetProjectLocator.findProjectWithImport(
-      '@proj/proj123-base',
-      'libs/proj/index.ts'
-    );
-    expect(similarImportFromNpm).toEqual('npm:@proj/proj123-base');
-
-    const similarDeepImportFromNpm = targetProjectLocator.findProjectWithImport(
-      '@proj/proj123-base/deep',
-      'libs/proj/index.ts'
-    );
-    expect(similarDeepImportFromNpm).toEqual('npm:@proj/proj123-base');
-  });
+jest.mock('nx/src/plugins/js/utils/find-external-package-json-path', () => ({
+  findExternalPackageJsonPath: jest
+    .fn()
+    .mockImplementation((packageName) =>
+      join('/root', 'node_modules', packageName, 'package.json')
+    ),
+}));
 
-  it('should return null for native modules', () => {
-    const result = targetProjectLocator.findProjectWithImport(
-      'path',
-      'libs/proj/index.ts'
-    );
-    expect(result).toEqual(null);
+describe('target-project-locator', () => {
+  describe('TargetProjectLocator', () => {
+    describe('findTargetProjectWithImport', () => {
```

Comment:
Oh boy, it looks like `git` thinks this whole file has changed... Is it because of the extra nesting here?

I imagine that isn't actually the case.. Is it possible for you to change the nesting here back so that git produces a reasonable diff please?